front/app/src/Views/GameCanvas/objects/Ball.tsx

```diff
@@ -0,0 +1,207 @@
+			{
+				const newPosX = this.pos.x + Math.cos(this.angle) * this.speedX;
+				const newPosY = this.pos.y
+								+ Math.sin(this.angle) * this.speedX;
+				const rp = this.game.playerTwo.isRightPlayer(newPosX, newPosY);
+				if (this.game.playerOne.isLeftPlayer(newPosX, newPosY) === true)
+				{
+					this.angle = Math.PI - this.angle;
+					this.angle = this.degreesToRadians(0);
+					this.angle -= this.degreesToRadians(this.getBounceAngle());
+				}
+				else if (rp === true)
+				{
+					this.pos.x = this.game.playerTwo.pos.x - this.radius;
+					this.angle = Math.PI - this.angle;
+					this.angle += this.degreesToRadians(this.getBounceAngle());
+				}
+				else
+				{
+					this.pos.x += Math.cos(this.angle) * this.speedX;
+					this.pos.y += Math.sin(this.angle) * this.speedX;
+				}
+				this.wasTopWallHit();
+				this.wasBottomWallHit();
+				if (this.pos.x <= 0)
+				{
+					this.game.playerTwo.score += 1;
+					// this.failedAudio.play();
+					this.angle = this.degreesToRadians(180);
+					this.init();
+				}
+				if (this.pos.x >= this.game.board.dim.width)
+				{
+					this.game.playerOne.score += 1;
+					// this.failedAudio.play();
+					this.angle = this.degreesToRadians(0);
+					this.init();
+				}
+			}
+		};
+		this.wasTopWallHit = () =>
+		{
+			if ( this.pos.y < this.radius )
+			{
+				// this.touchAudio.play();
+				this.pos.y = this.radius;
+				this.angle = ( Math.PI * 2 ) - this.angle;
+			}
+		};
+		this.wasBottomWallHit = () =>
+		{
+			if (this.game)
+			{
+				if (this.pos.y + this.radius > this.game.board.dim.height)
+				{
+					// this.touchAudio.play();
+					this.pos.y = this.game.board.dim.height - this.radius;
+					this.angle = ( Math.PI * 2 ) - this.angle;
+				}
+			}
+		};
+		this.degreesToRadians = (degrees) =>
+		{
+			const pi = Math.PI;
+			return (degrees * (pi/180));
+		};
+		this.radiansToDegrees = (radians) =>
+		{
+			const pi = Math.PI;
+			return (radians * (180/pi));
+		};
+		this.getBounceAngle = () =>
+		{
+			let relativeIntersectY;
+			relativeIntersectY = 0;
+			if (this.game)
+			{
+				if (this.pos.x > this.game.board.dim.width / 2)
+					relativeIntersectY = (this.game.playerTwo.pos.y
+						+ (this.game.playerTwo.racket.dim.height / 2))
+						- this.pos.y;
+				else
+					relativeIntersectY = (this.game.playerOne.pos.y
+						+ (this.game.playerTwo.racket.dim.height / 2))
+						- this.pos.y;
+			}
+			// ball's interception relative to the middle of the paddle
+			let normalizedRelativeIntersectionY;
+			if (this.game)
+				normalizedRelativeIntersectionY = relativeIntersectY
+					/ (this.game.playerTwo.racket.dim.height / 2);
+			let bounceAngle;
+			bounceAngle = 0;
+			if (normalizedRelativeIntersectionY)
+			{
+				bounceAngle = normalizedRelativeIntersectionY
+					* this.radiansToDegrees((Math.PI / 4));
+				if (normalizedRelativeIntersectionY < 0)
+					normalizedRelativeIntersectionY *= -1;
+				if (normalizedRelativeIntersectionY <= 0.15)
```

Comment:
This part must be fixed, maybe on a dedicated PR,
I think we have acceleration when hit the paddle, 
and decelleration should be when paddle angle reflected is the oposite of this angle of the ball incomming. (with a gradient that apply this effect)
Example Incoming angle (not real value just for explanation) : 180 and reflected angle 0 the ball go the the same direction it come.
This example value is not realistic, just to have the opposite value. 
Maybe we can define what is the angle from the paddle that triger this feature and a gradient to have a coef from the center of the paddle to make less or more effect.